lib/logstash/plugin_mixins/aws_config.rb

```diff
@@ -56,25 +59,21 @@ def aws_options_hash
       @logger.warn("Likely config error: Only one of access_key_id or secret_access_key was provided but not both.")
     end
 
-    if ((!@access_key_id || !@secret_access_key)) && @aws_credentials_file
-      access_creds = YAML.load_file(@aws_credentials_file)
-
-      @access_key_id = access_creds[:access_key_id]
-      @secret_access_key = access_creds[:secret_access_key]
-    end
-
-    opts = {}
-
-    if (@access_key_id && @secret_access_key)
-      opts[:access_key_id] = @access_key_id
-      opts[:secret_access_key] = @secret_access_key
+    if @access_key_id && @secret_access_key
+      opts = {
+        :access_key_id => @access_key_id,
+        :secret_access_key => @secret_access_key
+      }
+      opts[:session_token] = @session_token if @session_token
+    elsif @aws_credentials_file
+      opts = YAML.load_file(@aws_credentials_file)
+    else
+      raise LogStash::ConfigurationError.new('Neither credential file and credential was provided')
```

Comment:
@nicholasren I was reviewing old tickets concerning the AWS and I did a double check on this PR.
the raise `LogStash::ConfigurationError` will break defining the AWS configuration in the shell environment. 

We could add a default to the config to load the configuration from the environment.

```ruby
config : access_key_id, :default => ENV['AWS_ACCESS_KEY_ID']
config : access_key_id, :default => ENV['AWS_SECRET_ACCESS_KEY']
config : access_key_, :default => ENV['AWS_SESSION_TOKEN']
```

Sadly we don't have a test to cover this case :(